packages/loader/container-loader/src/retriableDocumentStorageService.ts

```diff
@@ -89,17 +109,40 @@ export class RetriableDocumentStorageService implements IDocumentStorageService
                 if (!canRetryOnError(err)) {
                     throw err;
                 }
+                numRetries += 1;
+                lastError = err;
                 // If the error is throttling error, then wait for the specified time before retrying.
                 // If the waitTime is not specified, then we start with retrying immediately to max of 8s.
                 retryAfter = getRetryDelayFromError(err) ?? Math.min(retryAfter * 2, 8000);
+                totalWaitTime += retryAfter;
                 if (id === undefined) {
                     id = uuid();
                 }
                 this.deltaManager.emitDelayInfo(id, retryAfter, CreateContainerError(err));
                 await this.delay(retryAfter);
             }
         } while (!success);
+        if (numRetries > 0) {
+            this.logger.sendTelemetryEvent({
+                eventName: "StorageCallsRetried",
+                numRetries,
+                fetchCallName,
+                totalWaitTime,
+            },
+            lastError);
+        }
         // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
         return result!;
     }
 }
+
+enum FetchCallName {
```

Comment:
Can you please move it to the top of the file?

Reply:
That said, I do not think we need enum here. It adds to bundle size for nothing. Why not use free-form strings?

Reply:
done